sdk/search/azure-search-documents/tests/search_service_preparer.py

```diff
@@ -14,14 +14,53 @@
 import json
 import requests
 
-from devtools_testutils import AzureMgmtPreparer, ResourceGroupPreparer
+from devtools_testutils import AzureMgmtPreparer, AzureMgmtTestCase, FakeResource, ResourceGroupPreparer
 from devtools_testutils.resource_testcase import RESOURCE_GROUP_PARAM
 from azure_devtools.scenario_tests.exceptions import AzureTestError
 
-SERVICE_URL_FMT = "https://{}.search.windows.net/indexes?api-version=2019-05-06"
+CREATE_INDEX_URL_FMT = "https://{}.search.windows.net/indexes?api-version=2019-05-06"
+DELETE_INDEX_URL_FMT = "https://{}.search.windows.net/indexes/{}?api-version=2019-05-06"
 TIME_TO_SLEEP = 3
 
+
+class GlobalResourceGroupPreparer(AzureMgmtPreparer):
```

Comment:
Per our OOB discussion (echoing here for others/posterity) CachedResourceGroupPreparer should serve this purpose, with the one exception that it simply takes the first requested CachedResourceGroup to utilize for the rest instead of naming specifically via an env var.

To unblock this PR I'll be approving but I'd aim to unify these sooner rather than later to consolidate as much of the test framework logic as we can.  Let this be my gentle ping to have a backlog issue or something such that it doesn't fall off the bus/feel free to just resolve this.